### PR TITLE
Fix #287 and improve CS/readability of various parts of the code

### LINKED ISSUE #287: PHP FATAL ERROR use FOS\MessageBundle\Model\ThreadMetadata
Hi

with last merge, i have an error =>
`
PHP Fatal error:  Cannot use FOS\MessageBundle\Model\ThreadMetadata as ThreadMetadata because the name is already in use in MessageBundle/Entity/Thread.php on line 12`

Suspected commit => #286 
https://github.com/FriendsOfSymfony/FOSMessageBundle/commit/d33c18a601173592ded7ccae41e5a818612e4a60#diff-a365cf1835fe7fc524c52bdfcc452db8R10

env : PHP 7.0.10

### PR DESCRIPTION
**Pull request by tgalopin**

### CODE REVIEW

File: Entity/MessageMetadata.php [hunk 1]
```diff
@@ -30,8 +30,7 @@ public function getMessage()
     }
 
     /**
-     * @param  MessageInterface $message
-     * @return null
+     * @param MessageInterface $message
```
Comment (Ener-Getick) [inline Entity/MessageMetadata.php]: This doc bloc brings no value

File: Entity/Thread.php [hunk 2]
```diff
@@ -76,7 +75,7 @@ public function getParticipants()
      */
     protected function getParticipantsCollection()
     {
-        if ($this->participants == null) {
+        if (!$this->participants) {
```
Comment (Ener-Getick) [inline Entity/Thread.php]: `null ===` is faster and stricter

File: FormModel/ReplyMessage.php [hunk 3]
```diff
@@ -22,8 +22,7 @@ public function getThread()
     }
 
     /**
-     * @param  ThreadInterface $thread
-     * @return null
+     * @param ThreadInterface $thread
```
Comment (Ener-Getick) [inline FormModel/ReplyMessage.php]: Useless as well

File: Model/Thread.php [hunk 4]
```diff
@@ -144,48 +143,47 @@ public function getIsSpam()
...
+     * @param boolean
      */
     public function setIsSpam($isSpam)
     {
         $this->isSpam = (boolean) $isSpam;
     }
 
     /**
-     * @see FOS\MessageBundle\Model\ThreadInterface::addMessage()
+     * @see \FOS\MessageBundle\Model\ThreadInterface::addMessage()
```
Comment (Ener-Getick) [inline Model/Thread.php]: Isn't it worth adding a use statement ?

File: ModelManager/MessageManager.php [hunk 5]
```diff
@@ -18,8 +20,7 @@
     public function createMessage()
     {
         $class = $this->getClass();
-        $message = new $class;
 
-        return $message;
+        return new $class;
```
Comment (Ener-Getick) [inline ModelManager/MessageManager.php]: What about adding empty brackets?

File: ModelManager/ThreadManager.php [hunk 6]
```diff
@@ -18,8 +20,7 @@
     public function createThread()
     {
         $class = $this->getClass();
-        $commentThread = new $class;
 
-        return $commentThread;
+        return new $class;
```
Comment (Ener-Getick) [inline ModelManager/ThreadManager.php]: Same

File: Model/Thread.php (hunk 4 above)
Comment (tgalopin) [inline Model/Thread.php]: I don't really like adding use only for comments :/ .

Comment (Ener-Getick) [inline Model/Thread.php]: And using `{@inheritdoc}`?

Comment (tgalopin) [inline Model/Thread.php]: That's what I did (not pushed yet) :)

**Comment by tgalopin** [comment]
Updated (fix issues and add other improvements)

**Pull request merged**